### PR TITLE
Extended MDR documentation

### PR DESCRIPTION
**Pull request by clobrano**
Signed-off-by: Carlo Lobrano <clobrano@gredhat.com>

### CODE REVIEW

File: remediation/machine-deletion/configuration.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+title: Configuration
+parent: Machine Deletion
+grand_parent: Remediation
+nav_order: 2
+---
+
+# Machine Deletion Remediation Configuration
+Machine Deletion Remediation (MDR) uses a Custom Resource (CR) created by [Node Health Check](https://github.com/medik8s/node-healthcheck-operator#readme) (NHC) based on a specific template.
+Since NHC comes with Self Node Remediation as the default remediation mechanism, it needs to be configured to use MDR; hence the following is required:
+1. Load the YAML manifest of the [MDR template](https://github.com/medik8s/machine-deletion#example-crs)
```
Comment (razo7) [inline remediation/machine-deletion/configuration.md]: I am not sure if load is the right word I would use here. Maybe `Add`

Comment (clobrano) [reply remediation/machine-deletion/configuration.md]: I like `Add`. I will change it also in MDR Readme

Comment (slintes) [reply remediation/machine-deletion/configuration.md]: we do not load or add manifests, we create resources / CRs 🙂  I'd suggest something like:
`Create the MDR template CR using the provided [manifest](https://github.com/medik8s/machine-deletion#example-crs)`

File: remediation/machine-deletion/how-it-works.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+layout: default
+title: How It Works
+parent: Machine Deletion
+grand_parent: Remediation
+nav_order: 1
+---
+
+# How Machine Deletion Remediation works?
+Machine Deletion Remediation (MDR) uses the [Machine API](https://github.com/openshift/machine-api-operator#readme) to reprovision unhealthy nodes.
+It was tested with [NodeHealthCheck](https://github.com/medik8s/node-healthcheck-operator#readme) (NHC), which creates a Custom Resource for MDR once a node turns unhealty.
```
Comment (mshitrit) [inline remediation/machine-deletion/how-it-works.md]: Typo, unhealty -> unhealthy

File: remediation/machine-deletion/how-it-works.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+MDR follows the annotation on the Node to the associated Machine object, confirms that it has an owning controller (e.g. MachineSetController), and deletes it. Finally, the owning controller creates a replacement.
```
Comment (mshitrit) [inline remediation/machine-deletion/how-it-works.md]: Might worth explain what it means in this context.
IIRC checking for an owning controller is our way of verifying the node is a worker node and not a control-plane node.

File: remediation/machine-deletion/machine-deletion.md [hunk 4]
```diff
@@ -10,6 +10,4 @@ parent: Remediation
 Generally Available
 {: .label .label-green }
 
-This operator conforms to the External Remediation of [NodeHealthCheck](https://github.com/medik8s/node-healthcheck-operator#readme) and is designed to work with it in order to reprovision unhealthy nodes using the [Machine API](https://github.com/openshift/machine-api-operator#readme). 
-It looks for the associated Machine of an unhealthy Node, and deletes it.  
-Once the Machine CR has been deleted, the owning controller creates a replacement. 
+[Machine Deletion Remediation](https://github.com/medik8s/machine-deletion#readme) is an External Remediation of [Node Health Check](https://github.com/medik8s/node-healthcheck-operator#readme), designed to reprovision unhealthy nodes using the [Machine API](https://github.com/openshift/machine-api-operator#readme).
```
Comment (mshitrit) [inline remediation/machine-deletion/machine-deletion.md]: IIUC `External Remediation` is a term that is relevant in the context of MHC and not NHC

Comment (clobrano) [reply remediation/machine-deletion/machine-deletion.md]: What would be a correct definition? This comes directly from the [current documentation](https://www.medik8s.io/remediation/machine-deletion/machine-deletion/)

File: remediation/machine-deletion/how-it-works.md (hunk 2 above)
Comment (clobrano) [reply remediation/machine-deletion/how-it-works.md]: good catch

File: remediation/machine-deletion/how-it-works.md (hunk 3 above)
Comment (clobrano) [reply remediation/machine-deletion/how-it-works.md]: I'll expand that

Comment (slintes) [reply remediation/machine-deletion/how-it-works.md]: > IIRC checking for an owning controller is our way of verifying the node is a worker node and not a control-plane node

yes and no: this ensures that someone is recreating the machine when we delete it. Which usually is the case for worker machines on IPI clusters, AFAIK.

File: remediation/machine-deletion/machine-deletion.md (hunk 4 above)
Comment (mshitrit) [reply remediation/machine-deletion/machine-deletion.md]: Maybe something like:
 `Machine Deletion Remediation` is a remediator that can be used by `Node Health Check` ... :thinking:

Comment (clobrano) [reply remediation/machine-deletion/machine-deletion.md]: ok, simple and clean

**Pull request merged**